roles/k8s_best_practices_certsuite/tasks/pre-run.yml

```diff
@@ -0,0 +1,219 @@
+        kbpc_certsuite_image: "{{ kbpc_project_name }}:{{ kbpc_image_tag }}"
+
+    - name: Build the certsuite image locally
+      ansible.builtin.shell: |
+        set -x
+        podman build -t {{ kbpc_certsuite_image }} .
+      args:
+        chdir: "{{ kbpc_certsuite_dir }}"
+
+- name: Tasks when testing a cnf-certification-test stable version
+  when:
+    - kbpc_code_src_dir is not defined
+  block:
+    - name: Create temporary directory for certsuite repository
+      ansible.builtin.tempfile:
+        state: directory
+      register: _kbpc_git_dir
+
+    - name: Define variable to refer to the directory where certsuite code is placed
+      ansible.builtin.set_fact:
+        kbpc_certsuite_dir: "{{ _kbpc_git_dir.path }}/{{ kbpc_project_name }}"
+
+    # In
+    # https://quay.io/repository/testnetworkfunction/cnf-certification-test?tab=tags,
+    # there are two main tags to take into account:
+    # -latest: linked to the latest certsuite release.
+    # -unstable: related to the HEAD version of the main branch in the certsuite repository
+    # (https://github.com/test-network-function/cnf-certification-test).
+    # So, if a version is provided in kbpc_version, that version
+    # will be used. In case we are on the master branch (HEAD version),
+    # unstable tag is used.
+    - name: Create variable with certsuite version name
+      ansible.builtin.set_fact:
+        kbpc_version_image: "{{ (kbpc_version | regex_search('HEAD')) | ternary('unstable', kbpc_version) }}"
+
+    - name: Create variable with certsuite image name
+      vars:
+        kbpc_image_tag: "{{ kbpc_version_image }}-{{ kbpc_image_suffix }}"
+      ansible.builtin.set_fact:
+        kbpc_certsuite_image: "{{ kbpc_project_name }}:{{ kbpc_image_tag }}"
+
+    - name: Clone certsuite repository
+      ansible.builtin.git:
+        repo: "{{ kbpc_repository }}"
+        version: "{{ kbpc_version }}"
+        dest: "{{ kbpc_certsuite_dir }}"
+        force: true
+        # Fetch only the latest commit to reduce the data
+        # to transfer when cloning the large repo
+        depth: 1
+        # Reuse objects from an existing repository to speed up cloning
+        reference: "{{ kbpc_certsuite_dir }}"
+      # On RHEL8 git clone can sporadically fail with OpenSSL SSL_read:
+      # SSL_ERROR_SYSCALL, errno 104. This is a workaround to try cloning the
+      # repo multiple times.
+      register: _kbpc_gitref
+      retries: 3
+      delay: 30
+      until: not _kbpc_gitref.failed
+
+    # For HEAD version, the kbpc_support_image_version should be 'latest'. Else, check version.json file
+    # from certsuite repo
+    - name: Set kbpc_support_image_version variable
+      vars:
+        _kbpc_version_json_file: "{{ lookup('file', kbpc_certsuite_dir + '/version.json') | from_json }}"
+      ansible.builtin.set_fact:
+        kbpc_support_image_version: "{{ 'latest' if kbpc_version == 'HEAD' else _kbpc_version_json_file.partner_tag[1:] }}"
```

Comment:
nitpick, please ignore: you've been using the ternary filter throughout the code

Reply:
good catch, again this task is inherited from the previous cnf_cert role, fixed